rosette/src/Heap.cc

```diff
@@ -633,10 +633,10 @@ void* OldSpace::alloc(unsigned sz) {
     if (sz <= MaxFixedSize && fixedFreeLists[sz])
         return (void*)unlink(fixedFreeLists[sz]);
 
-    if (p = currentChunk->alloc(sz))
+    if ((p = currentChunk->alloc(sz)))
```

Comment:
Split into two lines:
```c
p = currentChunk->alloc(sz);
if (p) { return p; }
```
or add the "Assign and test" comment like you do below.

Reply:
Which is your preference?

Reply:
My preference is splitting it up.

Reply:
Please no assignments in `if` statements. It doesn't compile to anything more efficient and is hard to read.

Also, for stylistic reasons, I recommend writing it like this:

```
auto p = currentChunk->alloc(sz);
if (NULL != p) {
  return p;
}
```

Since NULL can never be an lvalue in an assignment, this statement will produce a compiler error if you accidentally use `=` instead of `==` or `!=`. It's a worthwhile habit to always place the constants on the left hand side of the comparison for this reason.

Reply:
**Final:** We will refactor code to remove assignments from `if` and `while` statements. This may require turning some `while` statements into `do{...} while()` statements because the assignment has to happen before the test at the top of the loop.

Reply:
**Done**